app/src/main/java/org/mifos/mobile/ui/beneficiary_detail/BeneficiaryDetailScreen.kt

```diff
@@ -0,0 +1,286 @@
+                        deleteBeneficiaryClicked = false
+                    }
+                }
+
+                else -> throw IllegalStateException("Undesired $uiState")
+            }
+        }
+    }
+}
+
+
+@Preview(showSystemUi = true)
+@Composable
+fun PreviewBeneficiaryDetailScreenLoading() {
+    val beneficiary = Beneficiary(
+        id = 1,
+        name = "Beneficiary  name",
+        accountNumber = "123456789",
+        clientName = "Client Name",
+        accountType = null,
+        transferLimit = 1000.0,
+        officeName = "Main Office"
+    )
+
+    MifosMobileTheme {
+        BeneficiaryDetailScreen(
+            beneficiary = beneficiary,
+            uiState = BeneficiaryUiState.Loading,
+            navigateBack = {},
+            updateBeneficiaryClicked = {},
+            deleteBeneficiary = {}
+        )
+    }
+}
+
+@Preview(showSystemUi = true)
+@Composable
+fun PreviewBeneficiaryDetailScreenInitial() {
+    val beneficiary = Beneficiary(
+        id = 1,
+        name = "Beneficiary  name",
+        accountNumber = "123456789",
+        clientName = "Client Name",
+        accountType = null,
+        transferLimit = 1000.0,
+        officeName = "Main Office"
+    )
+    MifosMobileTheme {
+        BeneficiaryDetailScreen(
+            beneficiary = beneficiary,
+            uiState = BeneficiaryUiState.Initial,
+            navigateBack = {},
+            updateBeneficiaryClicked = {},
+            deleteBeneficiary = {}
+        )
+    }
+}
+
+@Composable
+@Preview(showSystemUi = true)
+fun PreviewBeneficiaryDetailScreenDeletedSuccessfully(
+    modifier: Modifier = Modifier
+) {
+    val beneficiary = Beneficiary(
+        id = 1,
+        name = "Beneficiary  name",
+        accountNumber = "123456789",
+        clientName = "Client Name",
+        accountType = null,
+        transferLimit = 1000.0,
+        officeName = "Main Office"
+    )
+
+    MifosMobileTheme {
+        BeneficiaryDetailScreen(
+            beneficiary = beneficiary,
+            uiState = BeneficiaryUiState.DeletedSuccessfully,
+            navigateBack = {},
+            updateBeneficiaryClicked = {},
+            deleteBeneficiary = {}
+        )
+    }
+}
+
+@Composable
+@Preview(showSystemUi = true)
+fun PreviewBeneficiaryDetailScreenShowError(modifier: Modifier = Modifier) {
+    val beneficiary = Beneficiary(
+        id = 1,
+        name = "Beneficiary  name",
+        accountNumber = "123456789",
+        clientName = "Client Name",
+        accountType = null,
+        transferLimit = 1000.0,
+        officeName = "Main Office"
+    )
+    MifosMobileTheme {
+        BeneficiaryDetailScreen(
+            beneficiary = beneficiary,
+            uiState = BeneficiaryUiState.ShowError(R.string.error_deleting_beneficiary),
+            navigateBack = {},
+            updateBeneficiaryClicked = {},
+            deleteBeneficiary = {}
+        )
+    }
+}
```

Comment:
use preview parameters, you can take reference from some screeens or my latest PR